### PR TITLE
For #23966 - Migrate MessageCardViewHolder to Compose

### LINKED ISSUE #23966: Migrate MessageCardViewHolder to Compose
Breakage of https://github.com/mozilla-mobile/fenix/issues/22755

Refers to [MessageCardViewHolder](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt) which is used to show a card letting users easily set Fenix as the default browser.

The compose version can be added to the current RecyclerView but before doing so we should ensure there will be no performance regression introduced.
To profile the differences the steps from https://github.com/mozilla-mobile/fenix/issues/21854#issuecomment-1049392266 can be used.

### PR DESCRIPTION
**Pull request by sarah541**
Fixes #23966 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: .experimenter.yaml
Comment (gabrielluong) [inline .experimenter.yaml:36]: Can you undo this

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt:50]:
```suggestion
        val message by remember { mutableStateOf(messageGlobal) }
```

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/compose/MessageCard.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/MessageCard.kt:94]: This needs a rebase since we removed `PrimaryText` and `SecondaryText`

**Review by Amejia481 — COMMENTED** [review]
The PR looks good!

Lets wait until we land https://github.com/mozilla-mobile/fenix/pull/25447 as it could change way in which report the message shown event.

**Review by Amejia481 — APPROVED** [review]
No we are good to go, we just need to rebase.
Let's just keep in mind that we don't need [onMessageDisplayed](https://github.com/mozilla-mobile/fenix/pull/25531/files#diff-0b84251795996e7d7803a6ed5fa535e66a3108b748615bfc98e1be7ac5c7f0b7R45) any more in the `ViewHolder` as we are already doing it in the middleware.

**Pull request merged**
